diff --git a/build/ink.d.ts b/build/ink.d.ts
index 9dce0fe328a8eeb18d459b974ab0c3cf5fb38dda..c567a5d4c65057b5816cdb3a1c6d1a1879fa3435 100644
--- a/build/ink.d.ts
+++ b/build/ink.d.ts
@@ -4,6 +4,7 @@ export type Options = {
     stdin: NodeJS.ReadStream;
     stderr: NodeJS.WriteStream;
     debug: boolean;
+    preserveScrollback?: boolean;
     exitOnCtrlC: boolean;
     patchConsole: boolean;
     isScreenReaderEnabled?: boolean;
diff --git a/build/ink.js b/build/ink.js
index 5cad56e135f1074474d7d498ffd87c5699b9df78..d220614ce7ed42de5e3838931886432408367678 100644
--- a/build/ink.js
+++ b/build/ink.js
@@ -129,7 +129,7 @@ export default class Ink {
         if (hasStaticOutput) {
             this.fullStaticOutput += staticOutput;
         }
-        if (this.lastOutputHeight >= this.options.stdout.rows) {
+        if (!this.options.preserveScrollback && this.lastOutputHeight >= this.options.stdout.rows) {
             this.options.stdout.write(ansiEscapes.clearTerminal + this.fullStaticOutput + output + '\n');
             this.lastOutput = output;
             this.lastOutputHeight = outputHeight;
diff --git a/build/parse-keypress.js b/build/parse-keypress.js
index 1e13e819f59d259a3476510db822695576a9d93c..e100611d8661187c4ca191e89621cd5e480fcacc 100644
--- a/build/parse-keypress.js
+++ b/build/parse-keypress.js
@@ -161,9 +161,9 @@ const parseKeypress = (s = '') => {
         key.meta = s.charAt(0) === '\x1b';
     }
     else if (s === '\x7f' || s === '\x1b\x7f') {
-        // TODO(vadimdemedes): `enquirer` detects delete key as backspace, but I had to split them up to avoid breaking changes in Ink. Merge them back together in the next major version.
-        // delete
-        key.name = 'delete';
+        // On macOS, \x7f is what the backspace key sends, not delete
+        // The actual forward delete sends escape sequences like \x1b[3~
+        key.name = 'backspace';
         key.meta = s.charAt(0) === '\x1b';
     }
     else if (s === '\x1b' || s === '\x1b\x1b') {
diff --git a/build/render.d.ts b/build/render.d.ts
index ffcfbaca277cbd95b2e6a2c857230dd69e41b7e0..0b0a6b0ccc6a6c24b3b7ea673662fc1e23db6e25 100644
--- a/build/render.d.ts
+++ b/build/render.d.ts
@@ -24,6 +24,8 @@ export type RenderOptions = {
      * @default false
      */
     debug?: boolean;
+    /** Preserve prior terminal output when an Ink view exceeds terminal height. */
+    preserveScrollback?: boolean;
     /**
      * Configure whether Ink should listen to Ctrl+C keyboard input and exit the app. This is needed in case `process.stdin` is in raw mode, because then Ctrl+C is ignored by default and process is expected to handle it manually.
      *
